siocli/lib/SqlDump.cpp

```diff
@@ -0,0 +1,796 @@
+        case ColumnDataType::COLUMN_DATA_TYPE_BINARY: {
+            const auto binaryValue = readBinaryValue(input);
+
+            std::stringstream ss;
+            ss << "X'" << std::setfill('0') << std::hex;
+            for (std::size_t i = 0; i < binaryValue.size(); ++i) {
+                const std::uint16_t v = binaryValue[i];
+                ss << std::setw(2) << v;
+            }
+            ss << '\'';
+
+            return ss.str();
+        }
+        case ColumnDataType::COLUMN_DATA_TYPE_TIMESTAMP: {
+            std::stringstream ss;
+            siodb::RawDateTime dateTime;
+            if (!siodb::protobuf::readRawDateTime(input, dateTime))
+                throw std::runtime_error("Read timestamp value failed");
+
+            ss << '\'' << dateTime.formatDefault() << '\'';
+            return ss.str();
+        }
+        default: throw std::runtime_error("Unknown data type");
+    }
+}
+
+std::string formAlterTableSetNextTrid(
+        const std::string& databaseName, const std::string& tableName, std::uint64_t nextTrid)
+{
+    std::stringstream ss;
+    ss << "ALTER TABLE " << databaseName << '.' << tableName << " SET NEXT_TRID = " << nextTrid;
+    return ss.str();
+}
+
+void checkResponse(const client_protocol::ServerResponse& response)
+{
+    const auto messageCount = response.message_size();
+    if (messageCount > 0) {
+        bool sqlErrorOccurred = false;
+        for (int i = 0; i < messageCount; ++i) {
+            const auto& message = response.message(i);
+            sqlErrorOccurred |= message.status_code() != 0;
+        }
+
+        if (sqlErrorOccurred) throw std::runtime_error("SQL error");
+    }
+}
+
+client_protocol::ServerResponse sendCommand(
+        std::string&& command, io::IoBase& connectionIo, protobuf::CustomProtobufInputStream& input)
+{
+    static std::int64_t requestId = 0;
+    client_protocol::Command clientCommand;
+    clientCommand.set_request_id(requestId);
+    clientCommand.set_text(std::move(command));
+    protobuf::writeMessage(protobuf::ProtocolMessageType::kCommand, clientCommand, connectionIo);
+
+    client_protocol::ServerResponse response;
+    protobuf::readMessage(protobuf::ProtocolMessageType::kServerResponse, response, input);
+
+    checkResponse(response);
+    ++requestId;
+
+    return response;
+}
+
+std::vector<DatabaseInfo> dumpDatabasesList(
+        io::IoBase& connectionIo, std::ostream& os, protobuf::CustomProtobufInputStream& input)
+{
+    auto query = formSelectCoreBody(kSystemDatabaseName, kSysDatabasesTableName,
+            {kSysDatabases_Name_ColumnName, kSysDatabases_CipherId_ColumnName,
+                    kSysDatabases_CipherKey_ColumnName});
+    const auto response = sendCommand(std::move(query), connectionIo, input);
+
+    std::vector<DatabaseInfo> databases;
+    databases.reserve(16);
+
+    const int columnCount = response.column_description_size();
+    if (columnCount != 3) std::runtime_error("Unexpected SYS DATABASES column count");
```

Comment:
I can see that for some queries there is such check, and for some there is no. Is this check required? Please be consistent. Either put such check on all queries or remove it from all queries.